lib/streamlit/elements/arrow.py

```diff
@@ -87,6 +87,57 @@ def _arrow_dataframe(
             ),
         )
 
+    def experimental_data_grid(
+        self,
+        data: Data = None,
+        width: Optional[int] = None,
+        height: Optional[int] = None,
+    ) -> "streamlit.delta_generator.DeltaGenerator":
+        """Display a dataframe with our new interactive table component.
+
+        Parameters
+        ----------
+        data : pandas.DataFrame, pandas.Styler, pyarrow.Table, numpy.ndarray, Iterable, dict, or None
+            The data to display.
+            If 'data' is a pandas.Styler, it will be used to style its
+            underyling DataFrame.
+        width : int or None
+            Desired width of the UI element expressed in pixels. If None, a
+            default width based on the page width is used.
+        height : int or None
+            Desired height of the UI element expressed in pixels. If None, a
+            default height is used.
+        Examples
```

Comment:
formatting nit: would be a bit more readable having a newline between the sections (and also a newline between the two examples)